### PR TITLE
Sorting Invoices and Address fixes

### LINKED ISSUE #2740: Contact us - address separate by comma
**Screenshot:**
Screenshot 2023-10-19 at 2 28 31 PM

**Changes:**
Have one character gap after comma

### PR DESCRIPTION
**Pull request by Gurmeenkour99**
#2740 #2750 fixes

Adjusted address spacing and fixed the sorting based on date field.

### CODE REVIEW

**Review by sandesh556 — APPROVED** [review]

File: app/Http/Controllers/Order/InvoiceController.php [hunk 1]
```diff
@@ -152,6 +152,7 @@ public function getInvoices(Request $request)
         $from = $request->input('from');
         $till = $request->input('till');
         $query = $this->advanceSearch($name, $invoice_no, $currency, $status, $from, $till);
+        $query->orderBy('date', 'desc');
```
Comment (sowmiyasubramaniyan) [inline app/Http/Controllers/Order/InvoiceController.php:155]: I don't think the sorting arrows will work after adding the orderBy

**Review by sowmiyasubramaniyan — CHANGES_REQUESTED** [review]

File: app/Http/Controllers/Order/InvoiceController.php [hunk 2]
```diff
@@ -152,14 +152,15 @@ public function getInvoices(Request $request)
...
```
Comment (sowmiyasubramaniyan) [inline app/Http/Controllers/Order/InvoiceController.php:155]: Remove this

**Review by sowmiyasubramaniyan — CHANGES_REQUESTED** [review]

**Review by sowmiyasubramaniyan — APPROVED** [review]

File: app/Http/Controllers/Order/InvoiceController.php [hunk 3]
```diff
@@ -155,11 +155,11 @@ public function getInvoices(Request $request)
 
         return \DataTables::of($query)
          ->setTotalRecords($query->count())
-         ->orderColumn('number', '-invoices.id $1')
-         ->orderColumn('user_id', 'invoices.id $1')
-         ->orderColumn('date', 'invoices.id $1')
-         ->orderColumn('status', 'invoices.id $1')
-         ->orderColumn('grand_total', 'invoices.id $1')
+         ->orderColumn('number', '-invoices.date $1')
```
Comment (sandesh556) [inline app/Http/Controllers/Order/InvoiceController.php:158]: Lets discuss this once

**Review by sandesh556 — CHANGES_REQUESTED** [review]
I need to see test cases also

File: app/Http/Controllers/Order/InvoiceController.php (hunk 3 above)
Comment (Gurmeenkour99) [reply app/Http/Controllers/Order/InvoiceController.php:158]: Sure

**Comment by Gurmeenkour99** [comment]
@sandesh556 As per our discussion, I will begin reviewing the invoice orders in all the other sections.

File: app/Http/Controllers/Front/ClientController.php [hunk 4]
```diff
@@ -279,14 +279,14 @@ public function getInvoices()
...
-                    ->orderColumn('total', '-invoices.id $1')
-                    ->orderColumn('paid', '-invoices.id $1')
-                    ->orderColumn('balance', '-invoices.id $1')
-                    ->orderColumn('status', '-invoices.id $1')
-                    ->orderColumn('date', '-invoices.id $1')
+                    ->orderColumn('number', '-invoices.date ASC')
+                    ->orderColumn('orderNo', '-invoices.date ASC')
+                    ->orderColumn('date', '-invoices.date ASC')
+                    ->orderColumn('total', '-invoices.date ASC')
+                    ->orderColumn('pais', '-invoices.date ASC')
```
Comment (sowmiyasubramaniyan) [inline app/Http/Controllers/Front/ClientController.php:286]: paid or pais?

**Review by sowmiyasubramaniyan — CHANGES_REQUESTED** [review]

**Review by sowmiyasubramaniyan — APPROVED** [review]

**Review by sandesh556 — DISMISSED** [review]

**Comment by Sabaric12** [comment]
Round 1: Testing Summary
Tested this PR with the fresh installation 
The mentioned issue is working fine now 
Before Fix:
image
image
After Fix:
image
image
Total time in days: 1 day
Total issues: 0
Minor - 0 
Major - 0 
Critical - 0
No of times PR was sent for Functionality correction:0
PR breaks existing functionality: No

**Review by Sabaric12 — DISMISSED** [review]

**Review by ganthimathin — CHANGES_REQUESTED** [review]
I tested the sorting and Address space issues in existing DB.

Sorting issue fixed in Admin panel>Inovices table only.

In All users > select a user > in their invoices table - sorting is not working.
In Client panle > Inovices - sorting is not working.

Address space:

An extra comma , is showing in address field. - not fixed

In the attached image after address 4 (.) full stop is showing after that comma is added unncessarily.

[Invoice sorting not working in client panel.webm](https://github.com/ladybirdweb/agora-invoicing-community/assets/131641568/4031ef21-db3f-446d-b21c-19e9e894808d)

image

**Review by ganthimathin — APPROVED** [review]
Verified the mentioned issues in local-exiting DB. Issues are fixed.

Round 2 test summary:

Total time in days: 2 hours
Total issues: 2
Minor - 2
Major - 0
Critical - 0
No of times PR was sent for Functionality correction: 1
PR breaks existing functionality: No

Address comma issue fixed

**Pull request merged**